src/app.rb

```diff
@@ -0,0 +1,125 @@
+require './data/persistors/genre_persistor'
+require './data/persistors/music_album_persistor'
+require './data/persistors/book_persistor'
+require './data/persistors/label_persistor'
+require './data/persistors/item_persistor'
+require './data/persistors/author_persistor'
+require './data/persistors/game_persistor'
+
+require './src/classes/book'
+require './src/classes/music_album'
+require './src/classes/game'
+
+class App
+  def initialize
+    @genres = GenrePersistor.read_from_file
+    @labels = LabelPersistor.read_from_file
+    @authors = AuthorPersistor.read_from_file
+    @music_albums = MusicAlbumPersistor.read_from_file(@labels, @authors, @genres)
+    @books = BookPersistor.read_from_file(@labels, @authors, @genres)
+    @games = GamePersistor.read_from_file(@labels, @authors, @genres)
+    @menu_options = {
+      '1' => method(:list_all_books),
+      '2' => method(:list_all_music_albums),
+      '3' => method(:list_all_games),
+      '4' => method(:list_all_genres),
+      '5' => method(:list_all_labels),
+      '6' => method(:list_all_authors),
+      '7' => method(:add_book),
+      '8' => method(:add_music_album),
+      '9' => method(:add_game)
+    }
+  end
+
+  def run
+    @menu_options.each_with_index do |option, index|
+      puts "#{index + 1} - #{option[1].name.to_s.split('_').join(' ').capitalize}"
+    end
+    puts '0 - Exit'
+
+    option = gets.chomp.to_s
+    if @menu_options.key?(option)
+      puts ''
+      @menu_options[option].call
+      puts ''
+      run
+    elsif option == '0'
+      save_data
+      puts "\nThank you for using this app!\n\n"
+    else
+      puts "\nThat is not a valid option\n\n"
+      run
+    end
+  end
+
+  def list_all_books
+    @books.each_with_index do |book, index|
+      puts "(#{index + 1}) - #{book.genre.genre} - #{book.author.first_name} - #{book.label.title}"
+    end
+  end
+
+  def list_all_music_albums
+    @music_albums.each_with_index do |music_album, index|
+      puts "(#{index + 1}) - #{music_album.genre.genre} - #{music_album.author.first_name} - #{music_album.label.title}"
+    end
+  end
```

Comment:
- [ ] Kindly make sure your application is working as expected. Right now, when trying to exit the app it shows the below errors i.e. when clicking the "0" option

### Error after entering some data
![Screenshot from 2023-01-15 07-04-54](https://user-images.githubusercontent.com/83042742/212523844-c88f9df2-5e08-4a71-a71c-d76f9ebdcc65.png)

### Error without data
![Screenshot from 2023-01-15 06-58-14](https://user-images.githubusercontent.com/83042742/212523872-7cbdd9e1-11a9-4448-b7e6-694ba1a5d514.png)

Reply:
Done! 😄